utils/near-performance-metrics/src/stats.rs

```diff
@@ -0,0 +1,341 @@
+{
+    let now = Instant::now();
+    let result = f();
+
+    let took = now.elapsed();
+    if took > SLOW_CALL_THRESHOLD {
+        info!(
+            "SLOW {}:{} {:?} took: {}ms",
+            class_name,
+            TID.with(|x| *x.borrow()),
+            msg_name,
+            took.as_millis(),
+        );
+    }
+
+    STATS.write().unwrap().log(class_name, msg_name, 0, took);
+    result
+}
+
+#[allow(unused_variables)]
+pub fn performance_with_debug<F, Message, Result>(
+    class_name: &'static str,
+    msg: Message,
+    f: F,
+) -> Result
+where
+    F: FnOnce(Message) -> Result,
+    Message: Debug,
+{
+    #[cfg(not(feature = "performance_stats"))]
+    {
+        f(msg)
+    }
+
+    #[cfg(feature = "performance_stats")]
+    {
+        let now = Instant::now();
+        let msg_test = format!("{:?}", msg);
+        let result = f(msg);
+
+        let took = now.elapsed();
+        if took > SLOW_CALL_THRESHOLD {
+            let msg_length = msg_test.len();
+            let mut msg_test = msg_test;
+            if msg_length >= MESSAGE_LIMIT {
+                msg_test = msg_test.as_str()[..MESSAGE_LIMIT].to_string();
+            }
+            info!(
+                "SLOW {}:{} {:?} took: {}ms len: {} {}",
+                class_name,
+                TID.with(|x| *x.borrow()),
+                std::any::type_name::<Message>(),
+                took.as_millis(),
+                msg_length,
+                msg_test
+            );
+        }
+
+        STATS.write().unwrap().log(class_name, std::any::type_name::<Message>(), 0, took);
+        result
+    }
+}
+
+pub struct MyFuture<F>
+where
+    F: futures::Future<Output = ()> + 'static,
+{
+    f: F,
+    class_name: &'static str,
+    file: &'static str,
+    line: u32,
+}
+
+impl<F> futures::Future for MyFuture<F>
+where
+    F: futures::Future<Output = ()> + 'static,
+{
+    type Output = ();
+
+    fn poll(self: Pin<&mut Self>, cx: &mut Context<'_>) -> Poll<Self::Output> {
+        let this = unsafe { self.get_unchecked_mut() };
+
+        let now = Instant::now();
+        let res = unsafe { Pin::new_unchecked(&mut this.f) }.poll(cx);
+        let took = now.elapsed();
+        STATS.write().unwrap().log(this.class_name, this.file, this.line, took);
+
+        if took > SLOW_CALL_THRESHOLD {
+            info!(
+                "SLOW {}:{} {}:{} took: {}ms",
+                this.class_name,
+                TID.with(|x| *x.borrow()),
+                this.file,
+                this.line,
+                took.as_millis()
+            );
+        }
+        match res {
+            Poll::Ready(x) => {
+                *REF_COUNTER.write().unwrap().entry((this.file, this.line)).or_insert_with(|| 0) -=
+                    1;
+                Poll::Ready(x)
+            }
+            Poll::Pending => Poll::Pending,
+        }
+    }
+}
+
+#[allow(unused_variables)]
+pub fn spawn<F>(class_name: &'static str, file: &'static str, line: u32, f: F)
```

Comment:
How do you feel about moving these helpers into a separate module? It looks a bit odd when you see `metrics::stats::spawn`.

Reply:
I'll change those function to be `metrics::actix::spawn`.